File: server/src/twine_ingest/ChildLink.py
class ChildLink:

    def __init__(self, child_data):
        """

        Parameters:
        -----------
        link_text (str): Text of the links to the child
        child_node (str): The unique page-id of the child node
        """
        self.link_text = child_data['link-text']
        self.child_id = child_data['child-ID']
        self.child_name = child_data['child-name']

    def updateLinkText(self, link_text):
        if type(link_text) is not str:
            return -1

        self.link_text = link_text
        return 0

    def toDict(self):
        return {'child-ID': self.child_id, 'child-name': self.child_name, 'link-text': self.link_text}


''' End Class '''

File: static/scripts/editor/ChildLink.js
export class ChildLink {
    constructor(child_id, child_name, link_text, ) {
        this.child_id = child_id;
        this.child_name = child_name;
        this.link_text = link_text;
    }

    updateLinkText(link_text) {
        self.link_text = str(link_text);
        return
    }

    toJSON() {
        return {
            'child_id': this.child_id,
            'child_name': this.child_name,
            'link_text': this.link_text
        }
    }
}